dwa_local_planner/src/dwa_planner_ros.cpp

```diff
@@ -429,6 +436,9 @@ namespace dwa_local_planner {
     }
 
     if (result == mbf_msgs::ExePathResult::SUCCESS) {
+      if (backwardDistanceExceeded(current_pose_, cmd_vel.twist.linear.x)) {
+        return mbf_msgs::ExePathResult::NO_VALID_CMD;
```

Comment:
That seems pretty strange to me; the planner might have found a forward command, but returned a backwards trajectory as the one with lowest cost?

Ideally, it would be the other way round: Once the robot traveled backwards too much, we should reject backwards trajectories (-> maybe as a critic?)

Another general concern is that this sort of logic will likely conflict with the `OscillationCostFunction` critic, which forces the planner to stick to the selected moving direction for some distance

Reply:
mmm...  maybe can extend the functionality of `PreferForwardCostFunction`:
- pass it the value of `max_backward_dist`
- move there the logic from `backwardDistanceExceeded`
- return inf cost whenever we are exceeding `max_backward_dist`

sounds reasonable?